File: claude-demo/frontend/src/api/tasks.ts
import type { Task, TaskCreate, TaskUpdate } from '../types'
import client from './client'

export const tasksApi = {
  list: () => client.get<Task[]>('/tasks/').then((r) => r.data),
  get: (id: string) => client.get<Task>(`/tasks/${id}`).then((r) => r.data),
  create: (data: TaskCreate) => client.post<Task>('/tasks/', data).then((r) => r.data),
  update: (id: string, data: TaskUpdate) =>
    client.put<Task>(`/tasks/${id}`, data).then((r) => r.data),
  delete: (id: string) => client.delete(`/tasks/${id}`),
}

File: claude-demo/app/routers/tasks.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List

from .. import crud, schemas
from ..database import get_db

router = APIRouter(prefix="/tasks", tags=["tasks"])


@router.get("/", response_model=List[schemas.TaskResponse])
def list_tasks(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    return crud.get_tasks(db, skip=skip, limit=limit)


@router.get("/{task_id}", response_model=schemas.TaskResponse)
def get_task(task_id: int, db: Session = Depends(get_db)):
    task = crud.get_task(db, task_id)
    if task is None:
        raise HTTPException(status_code=404, detail="Task not found")
    return task


@router.post("/", response_model=schemas.TaskResponse, status_code=201)
def create_task(task: schemas.TaskCreate, db: Session = Depends(get_db)):
    return crud.create_task(db, task)


@router.patch("/{task_id}", response_model=schemas.TaskResponse)
def update_task(task_id: int, task: schemas.TaskUpdate, db: Session = Depends(get_db)):
    updated = crud.update_task(db, task_id, task)
    if updated is None:
        raise HTTPException(status_code=404, detail="Task not found")
    return updated


@router.delete("/{task_id}", response_model=schemas.TaskResponse)
def delete_task(task_id: int, db: Session = Depends(get_db)):
    deleted = crud.delete_task(db, task_id)
    if deleted is None:
        raise HTTPException(status_code=404, detail="Task not found")
    return deleted
